### ISSUE TITLE
How do I know which field has blurred in the `asyncValidate` function?

### ISSUE THREAD

**Issue by techird** [erikras/redux-form#817]
I just can't find a proper way to do it. I don't want all my async fields being validated when there is just one field blurred.

**Comment by erikras** [comment]
Hmm.. I don't think that's made available to you. It's available to your form, via the `asyncValidating` prop, but not to your async validation function. It was assumed that you would probably only be validating a few fields, and they might depend on other fields, so you'd need all the form values.

**Comment by techird** [comment]
@erikras Yes, I agree with you. However,  in some scenario I need this, expecially I want to validate a field list in async way.

**Comment by Coobaha** [comment]
@techird  as a workaround, you can get private field with asyncValidated from form reducer

**Comment by techird** [comment]
@Coobaha That's a way out. And I am also asking for a better way that doesn't hacks.

**Comment by chelzwa** [comment]
Perhaps asyncValidate could receive the field which was blurred in addition to all the form values? I can definitely work around this, but it would be very helpful to know which field was blurred when running async validators.

**Comment by erikras** [comment]
> Perhaps asyncValidate could receive the field which was blurred in addition to all the form values? I can definitely work around this, but it would be very helpful to know which field was blurred when running async validators.

That seems very reasonable.

**Comment by kristian-puccio** [comment]
don't we already have this?
```
function validate(values, dispatch, props) {
const triggeredBy = props.form._active;
}
```

**Comment by erikras** [comment]
@kristian-puccio is correct. That is available, but it feels a little hacky.

**Comment by chelzwa** [comment]
@erikras Honestly, although props.form._active definitely does feel weird, it also does work pretty well for my use case.

**Comment by thomasmery** [comment]
Hi there,

as I am relying on the `props.form._active` hack to get which field just got blurred I'm facing a behaviour where `props.form._active` is `undefined` every second field that gets blurred

would someone tell me if they have seen this or if it rings a bell about something I might be doing wrong?

thnks

**Comment by mjward** [comment]
I'm also getting `undefined` when accessing `props.form._active`

**Comment by KaboomFox** [comment]
V6 doesn't have this private variable.

**Comment by KaboomFox** [comment]
it seems like a lot of people are having this issue [https://github.com/erikras/redux-form/issues/980](url) it is simple to add this to V6 but it requires you to combine old and new errors together before rejecting the promise (in your asyncValidate function). It would be nice the asyncValidate function just returned an object of errors so that asyncValidation function could contain the logic to combine new and old errors and reject/resolve the promise. Does this sound like a good idea? @erikras

**Comment by erikras** [comment]
@KaboomFox's fix published in [`v6.0.0-rc.5`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.5).
